### ISSUE TITLE
MailPoet redirects to its own landing page whenever any plugin is activated

### ISSUE THREAD

**Issue by filipecsweb** [mailpoet/mailpoet#5014]
**Describe the bug:**

The method `redirectToLandingPage` is exiting the code earlier, the method - `deleteOption` - isn't getting reached and the option value for "mailpoet_plugin_activated" is persiting.
That behavior makes the redirect to "/wp-admin/admin.php?page=mailpoet-landingpage" occurs whenever any plugin is activated because the option "mailpoet_plugin_activated" was never deleted.

**To reproduce:**

1. On a new WordPress installation install (don't activate yet) the plugins MailPoet and WP-Sweep (WP-Sweep is a good example because it doesn't redirect the user after it got activated).
2. Activate MailPoet, immediately leave the the landing page, go to the Plugins page and activate WP-Sweep.
3. You'll be redirected again to MailPoet's landing page.

**Expected behavior:**

I believe you should refactor the code and make sure the option "mailpoet_plugin_activated" is deleted.
Instead of:
```
$this->changelog->redirectToLandingPage();

// done with afterPluginActivation actions
$this->wpFunctions->deleteOption(self::PLUGIN_ACTIVATED);
```

do (just an example, more refactoring would be nedded):
```
if($this->changelog->redirectToLandingPage()) {
   $this->wpFunctions->deleteOption(self::PLUGIN_ACTIVATED);
   exit;
}
```
**Versions (please complete the following information):**

- WordPress version: 6.2.2
- PHP version: 8.1
- MailPoet version: 4.19.0

**Comment by rodrigoprimo** [comment]
Thanks for reporting this issue, @filipecsweb. I was able to reproduce this problem. We will fix it as soon as possible. Likely in the release that will go out next week.

**Comment by johnolek** [comment]
FYI @filipecsweb, the fix for this just went out in release [4.20.0](https://github.com/mailpoet/mailpoet/releases/tag/4.20.0). Thanks again for the report!
